plugins/window-rules/view-action-interface.cpp

```diff
@@ -342,6 +407,30 @@ void view_action_interface_t::_set_geometry(int x, int y, int w, int h)
     _move(x, y);
 }
 
+void view_action_interface_t::_set_geometry_ppt(int x, int y, int w, int h)
+{
+    auto output = _view->get_output();
+    if (!output)
+    {
+        return;
+    }
+
+    auto og = output->get_relative_geometry();
+
+    x = std::clamp(x, 0, 100);
+    y = std::clamp(y, 0, 100);
+    w = std::clamp(w, 0, 100);
+    h = std::clamp(h, 0, 100);
+
+    x = (og.width / 100) * x;
```

Comment:
Why not use double/float/(multiply first and then divide) for more precision here?

Reply:
Good idea, will update.